gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -43,7 +47,7 @@
  */
 public class SftpFsHelper implements SizeAwareFileBasedHelper {
   private static Logger log = LoggerFactory.getLogger(SftpFsHelper.class);
-  private ChannelSftp channelSftp;
+  Set<Channel> channels = Sets.newConcurrentHashSet();
```

Comment:
See my comments below. I think we should be using a channel pool and try to reuse open channels. At the very least these should weak references so those get collected asap instead of staying in memory and using up resources.